### PR TITLE
 Add section on LibEnsemble

### PR DESCRIPTION
**Pull request by MaxThevenet**
This PR documents how to use LibEnsemble on WarpX simulations. It contains:
- A new folder `Tools/LibEnsemble/` containing all scripts required to run LibEnsemble locally or on Summit.
- The corresponding section in the documentation, located in `Docs/source/libensemble/`.

### CODE REVIEW

File: Docs/source/index.rst [hunk 1]
```diff
@@ -41,3 +41,4 @@ In order to learn to use the code, please see the sections below:
    developers/doxygen
    acknowledge_us
    acknowledgements
+   libensemble/libensemble
```
[the remark this answers is not in the document]
Comment (MaxThevenet) [reply Docs/source/index.rst:44]: Agreed, I moved it before the acknowledgement sections, which should be slightly better.

**Comment by MaxThevenet** [comment]
Thanks for your review, I included all comments.

File: Docs/Doxyfile [hunk 2]
```diff
@@ -790,7 +790,7 @@ WARN_LOGFILE           =
 # spaces. See also FILE_PATTERNS and EXTENSION_MAPPING
 # Note: If this tag is empty the current directory is searched.
 
-INPUT                  = ../Source/
```
Comment (ax3l) [inline Docs/Doxyfile:793]: I was confused why we add the Tools here for doxygen source parsing. Are there C++ files in `Tools/` that we want to pick up?

Comment (ax3l) [reply Docs/Doxyfile:793]: clarified offline: Doxygen can actually parse the python scripts in `Tools/LibEnsemble` so we can add the `run_warpx` function doc string here.

**Pull request merged**